File: 프로그래머스/0/181844. 배열의 원소 삭제하기/배열의 원소 삭제하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<int> delete_list) {
    vector<int> answer;
    vector<int> tmp;
    for(int i=0; i<arr.size(); i++) {
        bool del = false;
        for(int j=0; j<delete_list.size(); j++) {
            if (arr[i] == delete_list[j]) {
                del = true;
                break;
            }
        }
        if (del) tmp.push_back(arr[i]);
    }
    for(int i: arr) {
        int check = false;
        for (int j: tmp) {
            if (i == j) check = true;
        }
        if (!check) answer.push_back(i);
    }
    return answer;
}

File: 프로그래머스/0/181844. 배열의 원소 삭제하기/배열의 원소 삭제하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[] delete_list) {
        int[] answer = new int[100];
        List<Integer> tmp = new ArrayList<>();
        for(int i: delete_list) {
            for(int j=0; j<arr.length; j++) {
                if (i == arr[j]) {
                    tmp.add(i);
                    break;
                }
            }
        }
        // for(int i:tmp) {
        //     System.out.println(i+" ");
        // }
        int k=0;
        for(int i=0; i<arr.length; i++) {
            boolean flag = true;
            for(int j=0; j<tmp.size(); j++) {
                if(arr[i] == tmp.get(j)) {
                    flag = false;
                    break;
                }
            }
            if(flag) {
                answer[k] = arr[i];
                k++;
            }
        }
        return Arrays.copyOf(answer, k); // 필요한 만큼만 배열 복사
    }
}